splunk-on-ubuntu/createUiDefinition.json

```diff
@@ -133,13 +90,363 @@
                 ]
             },
             {
-                "name": "splunkSettings",
-                "label": "Splunk settings",
+                "name": "nodesSettings",
+                "label": "Nodes Settings",
+                "subLabel": {
+                    "preValidation": "Required",
+                    "postValidation": "Done"
+                },
+                "bladeTitle": "Nodes Settings",
+                "elements": [
+                    {
+                        "name": "deploymentType",
+                        "type": "Microsoft.Common.DropDown",
+                        "label": "Splunk deployment type",
+                        "defaultValue": "Single node",
+                        "toolTip": "Single node deployment will deploy all Splunk components on a single VM, suitable for dev and test. Cluster mode deployment will deploy 1 Cluster Master VM, 3 Indexer VMs and 1 Search Head VM",
+                        "constraints": {
+                            "allowedValues": [
+                                {
+                                    "label": "Single node",
+                                    "value": "Standalone"
+                                },
+                                {
+                                    "label": "Cluster",
+                                    "value": "Cluster"
+                                }
+                            ],
+                            "required": true
+                        }
+                    },
+                    {
+                        "name": "clusterMasterVmSize",
+                        "type": "Microsoft.Compute.SizeSelector",
+                        "label": "Cluster Master VM size",
+                        "toolTip": "Select VM Size (will use 2 data disks)",
+                        "recommendedSizes": [
+                            "Standard_F4"
+                        ],
+                        "constraints": {
+                            "required": "[equals(steps('nodesSettings').deploymentType, 'Cluster')]",
+                            "allowedSizes": [
```

Comment:
have you considered using excludedSizes instead of allowedSizes? might help to shorten the list.